### ISSUE TITLE
Shrinking penalty

### ISSUE THREAD

**Issue by oktokat** [STAT540-UBC/Discussion#171]
Hi! This is probably the result of a rudimentary stats background, so I'm a little confused on the shrinking penalty applied in ridge regression.

I get that β̂ ridge=(X′X+λIp)^−1 * X′Y , so in this case it makes sense that increasing lambda would make β̂ ridge go closer to zero.

But I'm a little confused on how that equation fits into this one: RSS + λ∑βj^2

My main question is - how is a bigger lambda creating a smaller penalty overall? Does β̂ ridge =  λ∑βj^2 ?

Thanks!

**Comment by rbalshaw** [comment]
Hi @oktokat  -- You are on the right track, but your details are a bit off for the nature of the bias in the ridge-based estimators.

To answer your question, note that a bigger penalty means that when you try to minimize RSS + λ∑βj^2, solutions that have smaller values of all the βj values will be "better".  

That is, the ridge-based estimates of the betas will all be "shrunk" towards zero (i.e., smaller in absolute value).

See slides 11-17 (I think) in this lecture set.

http://statweb.stanford.edu/~tibs/sta305files/Rudyregularization.pdf

Sorry -- the slides are a bit dense, but on slide 11 it restates things, and then on slide 18 it gives the relationship you ask about.

Let me know if that doesn't untangle things for you.

**Comment by singha53** [comment]
@oktokat 

> I get that β̂ ridge=(X′X+λIp)^−1 * X′Y , so in this case it makes sense that increasing lambda would make β̂ ridge go closer to zero.
> But I'm a little confused on how that equation fits into this one: RSS + λ∑βj^2

you get β̂ ridge=(X′X+λIp)^−1 * X′Y by minimizing RSS + λ∑βj^2.

# Let's look at β̂LS (coefficients for least squares):
![image](https://cloud.githubusercontent.com/assets/8978305/24162148/d7a5a40a-0e23-11e7-9ad2-5ac619f7cf50.png)

# Let's look at β̂ridge (coefficients for ridge regression):
![image](https://cloud.githubusercontent.com/assets/8978305/24162321/5bf80748-0e24-11e7-8c9b-00593ff62045.png)

>My main question is - how is a bigger lambda creating a smaller penalty overall? Does β̂ ridge = λ∑βj^2 ?

you can see this from loss = RSS+λ∑βj^2 --> if you increase λ, the βj must decrease for a given loss. eg. suppose loss = 10 and λ = 100, then βj must be very tiny (and get tinnier as λ increases)  for this expression to hold.

You can also formulate this by constraining RSS subject to some equality constraint: see slide page 11 that @rbalshaw suggested. This can be solved using the method of Lagrange multipliers (https://en.wikipedia.org/wiki/Lagrange_multiplier). This is where slide 12 comes in for lecture 19: regularization.

**Comment by oktokat** [comment]
@singha53 @rbalshaw 
Ah, this makes so much more sense now! Thanks for the helpful ppt slides and the in-depth breakdown. Much appreciated!
